src/main.c

```diff
@@ -103,69 +84,82 @@ static void after_moving(void)
 }
 
 /**
- * @brief Includes the functions to be executed during exploration phase.
+ * @brief Let the user configure the mouse forces before exploring or running.
+ *
+ * The user selects the force to apply to the tires while exploring or running.
+ *
+ * @param[in] run Whether the robot should be running.
  */
-static void exploration_phase(void)
+static void configure_force(bool do_run)
 {
 	float force;
 
-	force = user_configuration(false);
+	force = hmi_configure_force(0.2, 0.05);
+	set_linear_speed_variables(force, do_run);
+
+	start_data_logging(log_data_control);
 	before_moving();
-	explore(force);
+	if (!do_run) {
+		explore(force);
+	} else {
+		run(force);
+		run_back(force);
+	}
 	after_moving();
+	stop_data_logging();
 }
 
 /**
- * @brief Includes the functions to be executed during the running phase.
+ * @brief Configure the goal for the search phase.
  */
-static void running_phase(void)
+static void configure_goal(void)
 {
-	float force;
-
-	force = user_configuration(true);
-	before_moving();
-	run(force);
-	run_back(force);
-	after_moving();
+	switch (button_user_wait_action()) {
+	case BUTTON_SHORT:
+		add_goal(1, 0);
+		break;
+	case BUTTON_LONG:
+		set_goal_classic();
+		speaker_play_competition();
+		break;
+	}
 }
 
 /**
- * @brief Competition routine.
+ * @brief Execute the exploration phase.
  */
-static void competition(void)
+static void exploration_phase(void)
 {
-	speaker_play_competition();
-	if (!reuse_maze()) {
-		initialize_solver_direction();
-		set_goal_classic();
-		set_target_goal();
-		exploration_phase();
-		set_run_sequence();
-		save_maze();
-		led_bluepill_on();
-	}
-	while (1)
-		running_phase();
+	set_search_initial_direction(NORTH);
+	configure_goal();
+	set_target_goal();
+
+	configure_force(false);
+
+	set_run_sequence();
+	save_maze();
 }
 
 /**
- * @brief Training routine.
+ * @brief Configure the robot and execute the search/run.
  */
-static void training(void)
+static void configure_and_solve(void)
 {
```

Comment:
Update the name to `configure_and_move`

Reply:
91cafc1eebca1275be5d81010a20b92c6e3963ff